src/vellum/workflows/nodes/bases/base.py

```diff
@@ -325,6 +326,36 @@ def should_initiate(
                 code=WorkflowErrorCode.INVALID_INPUTS,
             )
 
+        @classmethod
+        def _queue_node_execution(
+            cls, state: StateType, dependencies: Set["Type[BaseNode]"], invoked_by: Optional[Edge] = None
+        ) -> UUID:
+            """
+            Queues a future execution of a node, returning the span id of the execution.
+
+            We may combine this into the should_initiate method, but we'll keep it separate for now to avoid
+            breaking changes until the 0.15.0 release.
+            """
+
+            execution_id = uuid4()
+            if not invoked_by:
+                return execution_id
+
+            if cls.merge_behavior not in {MergeBehavior.AWAIT_ANY, MergeBehavior.AWAIT_ALL}:
+                return execution_id
```

Comment:
This is the only new piece of logic here. every thing else is copy pasted. The doc string for the method is also new